oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/resourceserver/web/BearerTokenAuthenticationFilter.java

```diff
@@ -83,20 +83,15 @@ protected void doFilterInternal(HttpServletRequest request, HttpServletResponse
 			return;
 		}
 
-		OAuth2ResourceAuthenticationToken authenticationRequest =
-			new OAuth2ResourceAuthenticationToken(token);
+		PreAuthenticatedAuthenticationToken authenticationRequest = new PreAuthenticatedAuthenticationToken(token, null);
 
 		authenticationRequest.setDetails(this.authenticationDetailsSource.buildDetails(request));
 
 		try {
-			OAuth2ResourceAuthenticationToken authenticationResult =
-				(OAuth2ResourceAuthenticationToken)
-					this.authenticationManager.authenticate(new OAuth2ResourceAuthenticationToken(token));
+			Authentication authenticationResult = this.authenticationManager.authenticate(authenticationRequest);
 
 			SecurityContextHolder.getContext().setAuthentication(authenticationResult);
 
-			onSuccessfulAuthentication(request, response, authenticationResult);
```

Comment:
I think it may be a mistake to not provide these hooks.

Insofar as these aren't related to changing the object hierarchy so it can support opaque tokens, would you mind putting this change in a different PR?

Reply:
> would you mind putting this change in a different PR

For sure, this should not be in this PR.

> I think it may be a mistake to not provide these hooks.

The one very important thing I would recommend to be aware of and to keep in the fore front of your mind is....do we need this now? If not than it goes. The more public API's we expose the more we need to support. The ultimate goal is to write minimal code that supports/provides the feature being implemented. I see this as a separate feature altogether and should be in a separate task. However, it would only get implemented if requested by the community. Until than I don't feel we should add this at this point.

Reply:
No problem. I'll probably log an issue separate from this release to identify the need to have hooks at various points of interest where folks can log, audit, and meter the authentication process.